app/models/portfolio.rb

```diff
@@ -8,7 +8,16 @@ class Portfolio < ApplicationRecord
   validates :image_url, :format => { :with => URI::DEFAULT_PARSER.make_regexp }, :allow_blank => true
   validates :enabled_before_type_cast, :format => { :with => /\A(true|false)\z/i }, :allow_blank => true
 
-  has_many :portfolio_items, :dependent => :destroy
+  has_many :portfolio_items, :dependent => :destroy do
+    def collect_with_regex(attr, regex)
```

Comment:
Hmm, I don't know if I like this method definition here. The method itself seems very cryptic (I mean, at first glance you open an empty array and then do `each` which I know is getting the `portfolio_items`, but again, at first glance it's confusing), and there is meta-programming in here that's going to make it very difficult to search for potential attribute method calls if we find something going wrong in debugging.

Also, you're only using this method in one single place, where the `each` part could easily just be inlined there to make it more readable.

Since it's only used in one place maybe it could be changed to simply collect by name so something like:
```ruby
def collect_with_regex_by_name(regex)
  map do |portfolio_item|
    portfolio_item.name if portfolio_item.name.match(regex) && !portfolio_item.name.nil?
  end
end
```

If not this, then at least we can probably use a `map` instead of `[].tap` and either don't put nils into the array, or do one of the following:
```ruby
map do |example|
  # ...
end.compact

# or

test = map do |example|
  # ...
end
test.compact
```

Reply:
Extracted the logic into a helper lib.